tools/utils/base_generator/main.py

```diff
@@ -0,0 +1,245 @@
+'''
+@file: main.py
+@author: MoWx 
+@brief:
+@copyright: 
+
+@attention:
+'''
+import os
+
+import os
+
+def get_license_header(license=None, max_line_length=80):
+    if license is None:
+        return None
+
+    license_header = []
+    
+    # Get the directory path where license files are stored
+    license_dir = os.path.join(os.getcwd(), 'licenses')
+    
+    # Check if the license file exists
+    if license not in os.listdir(license_dir):
+        raise ValueError('License file "{}" does not exist'.format(license))
+    
+    # Read the content of the license file and format each line
+    with open(os.path.join(license_dir, license), mode='r', encoding='utf-8') as fp:
+        for line in fp:
+            stripped_line = line.strip()
+            
+            # Skip empty lines and comments (lines starting with '#')
+            if len(stripped_line) > 0 and not stripped_line.startswith('#'):
+                # Limit the line length
+                segments = [stripped_line[i:i+max_line_length] for i in range(0, len(stripped_line), max_line_length)]
+                formatted_segments = []
+                for segment in segments:
+                    num_spaces = max_line_length - len(segment) + 3
+                    formatted_segment = '** {}{} **'.format(segment.strip(), ' ' * num_spaces)
+                    formatted_segments.append(formatted_segment)
+                formatted_line = '\n'.join(formatted_segments)
+                license_header.append(formatted_line)
+                
+    return '\n'.join(license_header) + '\n'
+            
+c_file_license = None
+if c_file_license is not None:
+    c_file_license_header = get_license_header(c_file_license, 71)
+
+    c_file_license_headerx = '''
+/*******************************************************************************
```

Comment:
To not start with empty line.
```suggestion
    c_file_license_headerx = '''/*******************************************************************************
```